src/blockdata/transaction.rs

```diff
@@ -397,10 +397,11 @@ impl Transaction {
         input_index: usize,
         script_pubkey: &Script,
         sighash_u32: u32
-    ) -> Result<SigHash, encode::Error> {
+    ) -> SigHash {
         let mut engine = SigHash::engine();
-        self.encode_signing_data_to(&mut engine, input_index, script_pubkey, sighash_u32)?;
-        Ok(SigHash::from_engine(engine))
+        self.encode_signing_data_to(&mut engine, input_index, script_pubkey, sighash_u32)
```

Comment:
I think the original `encode_signing_data_to` function should not return Error either, so is it better to apply expect there?

Reply:
It can f.e.e when the writer is a file or tcp stream that returns I/O errors. It just so happens that here we write into an `io::Write` that we know never returns I/O errors. Just like f.e. a `Vec<u8>` doesn't.